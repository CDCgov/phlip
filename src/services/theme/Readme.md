The project uses the material-ui library for React. Material-UI is based
on Google's Material Design. There's a global application theme of colors
and overrides that is given to the `<MuiThemeProvider>` component in
`src/App.js`.

To create the theme, the `createMuiTheme` function from the library is
used. We pass in the primary, secondary and error colors to the palette
property that can be referenced throughout the app on Material UI components.

Material UI allows you to override any default configuration. This project
has four overrides, defined in the `overrides` property:

- Form labels are set to the secondary color (teal) when focused
- Form labels are set to a gray color when disabled
- The underline of a text field when focus is the secondary color (teal)
- Avatar images are set to 100% width and 100% height, otherwise images
are made to fit the circle and may not fill the entire circle
