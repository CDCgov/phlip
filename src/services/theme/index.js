import { createMuiTheme } from '@material-ui/core/styles'
import { pink, deepPurple, teal } from '@material-ui/core/colors'

const mainColor = '#048484'

const theme = createMuiTheme({
  palette: {
    primary: {
      ...deepPurple,
      main: '#3d316a',
      light: '#6a5b98',
      dark: '#100b3f',
      500: '#3d316a'
    },
    secondary: {
      ...teal,
      main: mainColor,
      light: '#4fb4b4',
      dark: '#005757',
      pageHeader: '#005757',
      'A100': mainColor,
      'A200': mainColor, // MUI uses this color to determine color of buttons when set to 'secondary' (beta-25)
      'A400': '#005757', // MUI uses this color to determine color of buttons on hover when set to 'secondary' (beta-25)
      tabs: '#ecf7f6'
    },
    error: {
      ...pink,
      500: '#db036e',
      main: '#db036e'
    },
    greyText: '#757575'
  },
  scenes: {
    background: '#f5f5f5'
  },
  shape: {
    borderRadius: 2
  },
  overrides: {
    MuiFormLabel: {
      root: {
        '&$focused': {
          color: mainColor
        },
        '&$disabled': {
          color: 'black'
        }
      },
    },
    MuiInput: {
      underline: {
        '&:after': {
          borderBottom: `2px solid ${mainColor}`
        }
      }
    },
    MuiAvatar: {
      img: {
        width: '100%',
        height: '100%',
        textAlign: 'center',
        maxWidth: '100%'
      }
    }
  }
})

export default theme
