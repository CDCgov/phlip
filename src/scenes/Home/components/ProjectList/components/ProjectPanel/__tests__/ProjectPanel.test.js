import React from 'react'
import { shallow } from 'enzyme'
import { ProjectPanel } from '../index'
import theme from 'services/theme'

const props = {
  actions: {
    getProjectUsers: jest.fn(),
    toggleBookmark: jest.fn()
  },
  onExport: jest.fn(),
  length: 4,
  id: 1,
  index: 0,
  project: {
    createdBy: 'Admin',
    createdByEmail: 'admin@cdc.gov',
    createdById: 1,
    dateCreated: '2019-04-08T11:43:46.3034169',
    dateLastEdited: '2019-04-18T13:27:51.454041',
    id: 2,
    lastEditedBy: 'Aleksandra Zaryanova',
    lastUsersCheck: 1555606985567,
    name: 'PROJECT',
    projectUsers: [],
    projectJurisdictions: [],
    type: 1
  },
  handleExpandProject: jest.fn(),
  handleEditProject: jest.fn(),
  bookmarked: false,
  expanded: false,
  allUsers: {},
  role: 'Admin',
  users: [],
  theme
}

describe('Home - ProjectList - ProjectPanel scene', () => {
  test('should render correctly', () => {
    expect(shallow(<ProjectPanel {...props} />)).toMatchSnapshot()
  })
  
  test('should render correctly when expanded', () => {
    expect(shallow(<ProjectPanel {...props} expanded />)).toMatchSnapshot()
  })
  
  describe('when a coder is logged in', () => {
    const wrapper = shallow(<ProjectPanel {...props} expanded role="Coder" />)
    describe('when the card is open', () => {
      test('validate button should not be visible', () => {
        expect(wrapper.find('WithTheme(Button)').findWhere(n => n.prop('value') === 'Validate').length).toEqual(0)
      })
      
      test('code button should be visible', () => {
        expect(wrapper.find('WithTheme(Button)').findWhere(n => n.prop('value') === 'Code').length).toEqual(1)
      })
      
      test('coding scheme button should not be visible', () => {
        expect(wrapper.find('WithTheme(Button)').everyWhere(n => n.childAt(0) !== 'Coding Scheme')).toEqual(true)
      })
      
      test('jurisdictions button should not be visible', () => {
        expect(wrapper.find('WithTheme(Button)').everyWhere(n => n.childAt(0) !== 'Jurisdiction')).toEqual(true)
      })
    })
  })
  
  describe('when a coordinator is logged in', () => {
    const wrapper = shallow(<ProjectPanel {...props} expanded role="Coordinator" />)
    describe('when the card is open', () => {
      test('validate button should be visible', () => {
        expect(wrapper.find('WithTheme(Button)').findWhere(n => n.prop('value') === 'Validate').length).toEqual(1)
      })
      
      test('code button should be visible', () => {
        expect(wrapper.find('WithTheme(Button)').findWhere(n => n.prop('value') === 'Code').length).toEqual(1)
      })
      
      test('coding scheme button should be visible', () => {
        expect(wrapper.find('WithTheme(Button)').someWhere(n => n.childAt(0).text() === 'Coding Scheme')).toEqual(true)
      })
      
      test('jurisdictions button should be visible', () => {
        expect(wrapper.find('WithTheme(Button)').someWhere(n => n.childAt(0).text() === 'Jurisdictions')).toEqual(true)
      })
    })
  })
})
