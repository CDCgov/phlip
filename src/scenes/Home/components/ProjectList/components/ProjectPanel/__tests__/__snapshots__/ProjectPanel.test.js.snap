// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home - ProjectList - ProjectPanel scene should render correctly 1`] = `
<Fragment>
  <FlexGrid
    align="stretch"
    circular={false}
    container={false}
    flex={false}
    justify="stretch"
    padding={0}
    raised={false}
    style={Object {}}
    type="column"
  >
    <FlexGrid
      align="stretch"
      circular={false}
      container={true}
      flex={true}
      justify="center"
      padding={0}
      raised={false}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 0,
          "boxShadow": "0px 1px 3px 0px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
          "fontSize": "0.8125rem",
          "overflow": "hidden",
          "position": "relative",
        }
      }
      type="column"
    >
      <ProjectRow
        bookmarked={false}
        handleExpandProject={[Function]}
        isCoder={false}
        project={
          Object {
            "createdBy": "Admin",
            "createdByEmail": "admin@cdc.gov",
            "createdById": 1,
            "dateCreated": "2019-04-08T11:43:46.3034169",
            "dateLastEdited": "2019-04-18T13:27:51.454041",
            "id": 2,
            "lastEditedBy": "Aleksandra Zaryanova",
            "lastUsersCheck": 1555606985567,
            "name": "PROJECT",
            "projectJurisdictions": Array [],
            "projectUsers": Array [],
            "type": 1,
          }
        }
        toggleBookmark={[MockFunction]}
      />
      <WithStyles(Collapse)
        in={false}
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "transitionDuration": "150ms",
            "width": "100%",
          }
        }
      >
        <FlexGrid
          align="stretch"
          circular={false}
          container={true}
          flex={false}
          justify="stretch"
          padding={0}
          raised={false}
          style={
            Object {
              "overflow": "auto",
              "width": "100%",
            }
          }
          type="row"
        >
          <FlexGrid
            align="stretch"
            circular={false}
            container={true}
            flex={false}
            justify="stretch"
            padding={0}
            raised={false}
            style={
              Object {
                "backgroundColor": "white",
                "width": 300,
              }
            }
            type="column"
          >
            <WithStyles(GridList)
              cellHeight={300}
              cols={1}
              rows={1}
              spacing={4}
              style={
                Object {
                  "margin": 0,
                  "width": 300,
                }
              }
            >
              <WithStyles(GridListTile)
                cols={1}
                key="2-blank-0"
                rows={1}
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "#f9f9f9",
                      "height": 300,
                    }
                  }
                />
              </WithStyles(GridListTile)>
            </WithStyles(GridList)>
          </FlexGrid>
          <FlexGrid
            align="stretch"
            circular={false}
            container={true}
            flex={true}
            justify="stretch"
            padding={0}
            raised={false}
            style={Object {}}
            type="column"
          >
            <FlexGrid
              align="center"
              circular={false}
              container={true}
              flex={false}
              justify="stretch"
              onClick={[Function]}
              padding={30}
              raised={false}
              style={
                Object {
                  "backgroundColor": "#f9f9f9",
                }
              }
              type="row"
            >
              <FlexGrid
                align="center"
                circular={false}
                container={true}
                flex={true}
                justify="flex-start"
                padding={0}
                raised={false}
                style={
                  Object {
                    "fontSize": 20,
                    "minWidth": 150,
                  }
                }
                type="row"
              >
                <IconButton
                  aria-label="Bookmark this project"
                  color="#757575"
                  disabled={false}
                  iconSize={24}
                  id="bookmark-project-2"
                  onClick={[Function]}
                  tooltipText="Bookmark project"
                >
                  bookmark_border
                </IconButton>
                <FlexGrid
                  align="stretch"
                  circular={false}
                  container={false}
                  flex={false}
                  justify="stretch"
                  padding={0}
                  raised={false}
                  style={
                    Object {
                      "width": 20,
                    }
                  }
                  type="column"
                />
                <WithTheme(TextLink)
                  aria-label="Edit project details"
                  to={
                    Object {
                      "pathname": "/project/edit/2",
                      "state": Object {
                        "directEditMode": true,
                        "modal": true,
                        "projectDefined": Object {
                          "createdBy": "Admin",
                          "createdByEmail": "admin@cdc.gov",
                          "createdById": 1,
                          "dateCreated": "2019-04-08T11:43:46.3034169",
                          "dateLastEdited": "2019-04-18T13:27:51.454041",
                          "id": 2,
                          "lastEditedBy": "Aleksandra Zaryanova",
                          "lastUsersCheck": 1555606985567,
                          "name": "PROJECT",
                          "projectJurisdictions": Array [],
                          "projectUsers": Array [],
                          "type": 1,
                        },
                      },
                    }
                  }
                >
                  PROJECT
                </WithTheme(TextLink)>
              </FlexGrid>
              <FlexGrid
                align="stretch"
                circular={false}
                container={true}
                flex={true}
                justify="flex-end"
                padding={0}
                raised={false}
                style={
                  Object {
                    "height": 40,
                  }
                }
                type="row"
              >
                <WithTheme(Button)
                  aria-label="Add and edit project jurisdictions"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  id="2-edit-jurisdictions"
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to={
                    Object {
                      "pathname": "/project/2/jurisdictions",
                      "state": Object {
                        "modal": true,
                      },
                    }
                  }
                >
                  Jurisdictions
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Documents in this project"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to={
                    Object {
                      "pathname": "/docs",
                      "state": Object {
                        "project": Object {
                          "createdBy": "Admin",
                          "createdByEmail": "admin@cdc.gov",
                          "createdById": 1,
                          "dateCreated": "2019-04-08T11:43:46.3034169",
                          "dateLastEdited": "2019-04-18T13:27:51.454041",
                          "id": 2,
                          "lastEditedBy": "Aleksandra Zaryanova",
                          "lastUsersCheck": 1555606985567,
                          "name": "PROJECT",
                          "projectJurisdictions": Array [],
                          "projectUsers": Array [],
                          "type": 1,
                        },
                        "projectDefined": true,
                      },
                    }
                  }
                >
                  Documents
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Add and edit project coding scheme"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to="/project/2/coding-scheme"
                >
                  Coding Scheme
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Add and edit project protocol"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to="/project/2/protocol"
                >
                  Protocol
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Export validated questions"
                  color="white"
                  disableRipple={true}
                  id="export-validated-2"
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                >
                  Export
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
              </FlexGrid>
            </FlexGrid>
            <FlexGrid
              align="stretch"
              circular={false}
              container={true}
              flex={false}
              justify="stretch"
              padding={0}
              raised={false}
              style={Object {}}
              type="row"
            >
              <FlexGrid
                align="stretch"
                circular={false}
                container={true}
                flex={true}
                justify="stretch"
                padding={30}
                raised={false}
                style={Object {}}
                type="column"
              >
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Created by:
                     
                  </span>
                  Admin
                   -
                   
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": "0.8rem",
                      }
                    }
                  >
                    4/8/2019
                  </span>
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Last edited by:
                     
                  </span>
                  Aleksandra Zaryanova
                   -
                   
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": "0.8rem",
                      }
                    }
                  >
                    4/18/2019
                  </span>
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Coordinator(s):
                     
                  </span>
                  <span />
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Coder(s):
                     
                  </span>
                  <span />
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Type:
                     
                  </span>
                  <span>
                    Legal Scan
                  </span>
                </WithStyles(Typography)>
                <FlexGrid
                  align="stretch"
                  circular={false}
                  container={false}
                  flex={false}
                  justify="stretch"
                  padding={0}
                  raised={false}
                  style={Object {}}
                  type="column"
                />
              </FlexGrid>
              <FlexGrid
                align="flex-end"
                circular={false}
                container={true}
                flex={true}
                justify="flex-end"
                padding={30}
                raised={false}
                style={Object {}}
                type="row"
              >
                <WithTheme(Button)
                  aria-label="Code project"
                  component={[Function]}
                  listButton={true}
                  raised={false}
                  style={
                    Object {
                      "borderRadius": 3,
                      "height": 50,
                      "marginRight": 20,
                      "width": 150,
                    }
                  }
                  to={
                    Object {
                      "pathname": "/project/2/code",
                    }
                  }
                  value="Code"
                />
                <WithTheme(Button)
                  aria-label="Validate project"
                  component={[Function]}
                  listButton={true}
                  raised={false}
                  style={
                    Object {
                      "borderRadius": 3,
                      "height": 50,
                      "width": 150,
                    }
                  }
                  to={
                    Object {
                      "pathname": "/project/2/validate",
                    }
                  }
                  value="Validate"
                />
              </FlexGrid>
            </FlexGrid>
          </FlexGrid>
        </FlexGrid>
      </WithStyles(Collapse)>
    </FlexGrid>
  </FlexGrid>
</Fragment>
`;

exports[`Home - ProjectList - ProjectPanel scene should render correctly when expanded 1`] = `
<Fragment>
  <FlexGrid
    align="stretch"
    circular={false}
    container={false}
    flex={false}
    justify="stretch"
    padding={0}
    raised={false}
    style={Object {}}
    type="column"
  >
    <FlexGrid
      align="stretch"
      circular={false}
      container={true}
      flex={true}
      justify="center"
      padding={0}
      raised={false}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 2,
          "boxShadow": "0px 1px 3px 0px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
          "fontSize": "0.8125rem",
          "overflow": "hidden",
          "position": "relative",
        }
      }
      type="column"
    >
      <WithStyles(Collapse)
        in={true}
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "transitionDuration": "150ms",
            "width": "100%",
          }
        }
      >
        <FlexGrid
          align="stretch"
          circular={false}
          container={true}
          flex={false}
          justify="stretch"
          padding={0}
          raised={false}
          style={
            Object {
              "overflow": "auto",
              "width": "100%",
            }
          }
          type="row"
        >
          <FlexGrid
            align="stretch"
            circular={false}
            container={true}
            flex={false}
            justify="stretch"
            padding={0}
            raised={false}
            style={
              Object {
                "backgroundColor": "white",
                "width": 300,
              }
            }
            type="column"
          >
            <WithStyles(GridList)
              cellHeight={300}
              cols={1}
              rows={1}
              spacing={4}
              style={
                Object {
                  "margin": 0,
                  "width": 300,
                }
              }
            >
              <WithStyles(GridListTile)
                cols={1}
                key="2-blank-0"
                rows={1}
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "#f9f9f9",
                      "height": 300,
                    }
                  }
                />
              </WithStyles(GridListTile)>
            </WithStyles(GridList)>
          </FlexGrid>
          <FlexGrid
            align="stretch"
            circular={false}
            container={true}
            flex={true}
            justify="stretch"
            padding={0}
            raised={false}
            style={Object {}}
            type="column"
          >
            <FlexGrid
              align="center"
              circular={false}
              container={true}
              flex={false}
              justify="stretch"
              onClick={[Function]}
              padding={30}
              raised={false}
              style={
                Object {
                  "backgroundColor": "#f9f9f9",
                }
              }
              type="row"
            >
              <FlexGrid
                align="center"
                circular={false}
                container={true}
                flex={true}
                justify="flex-start"
                padding={0}
                raised={false}
                style={
                  Object {
                    "fontSize": 20,
                    "minWidth": 150,
                  }
                }
                type="row"
              >
                <IconButton
                  aria-label="Bookmark this project"
                  color="#757575"
                  disabled={false}
                  iconSize={24}
                  id="bookmark-project-2"
                  onClick={[Function]}
                  tooltipText="Bookmark project"
                >
                  bookmark_border
                </IconButton>
                <FlexGrid
                  align="stretch"
                  circular={false}
                  container={false}
                  flex={false}
                  justify="stretch"
                  padding={0}
                  raised={false}
                  style={
                    Object {
                      "width": 20,
                    }
                  }
                  type="column"
                />
                <WithTheme(TextLink)
                  aria-label="Edit project details"
                  to={
                    Object {
                      "pathname": "/project/edit/2",
                      "state": Object {
                        "directEditMode": true,
                        "modal": true,
                        "projectDefined": Object {
                          "createdBy": "Admin",
                          "createdByEmail": "admin@cdc.gov",
                          "createdById": 1,
                          "dateCreated": "2019-04-08T11:43:46.3034169",
                          "dateLastEdited": "2019-04-18T13:27:51.454041",
                          "id": 2,
                          "lastEditedBy": "Aleksandra Zaryanova",
                          "lastUsersCheck": 1555606985567,
                          "name": "PROJECT",
                          "projectJurisdictions": Array [],
                          "projectUsers": Array [],
                          "type": 1,
                        },
                      },
                    }
                  }
                >
                  PROJECT
                </WithTheme(TextLink)>
              </FlexGrid>
              <FlexGrid
                align="stretch"
                circular={false}
                container={true}
                flex={true}
                justify="flex-end"
                padding={0}
                raised={false}
                style={
                  Object {
                    "height": 40,
                  }
                }
                type="row"
              >
                <WithTheme(Button)
                  aria-label="Add and edit project jurisdictions"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  id="2-edit-jurisdictions"
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to={
                    Object {
                      "pathname": "/project/2/jurisdictions",
                      "state": Object {
                        "modal": true,
                      },
                    }
                  }
                >
                  Jurisdictions
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Documents in this project"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to={
                    Object {
                      "pathname": "/docs",
                      "state": Object {
                        "project": Object {
                          "createdBy": "Admin",
                          "createdByEmail": "admin@cdc.gov",
                          "createdById": 1,
                          "dateCreated": "2019-04-08T11:43:46.3034169",
                          "dateLastEdited": "2019-04-18T13:27:51.454041",
                          "id": 2,
                          "lastEditedBy": "Aleksandra Zaryanova",
                          "lastUsersCheck": 1555606985567,
                          "name": "PROJECT",
                          "projectJurisdictions": Array [],
                          "projectUsers": Array [],
                          "type": 1,
                        },
                        "projectDefined": true,
                      },
                    }
                  }
                >
                  Documents
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Add and edit project coding scheme"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to="/project/2/coding-scheme"
                >
                  Coding Scheme
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Add and edit project protocol"
                  color="white"
                  component={[Function]}
                  disableRipple={true}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                  to="/project/2/protocol"
                >
                  Protocol
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
                <WithTheme(Button)
                  aria-label="Export validated questions"
                  color="white"
                  disableRipple={true}
                  id="export-validated-2"
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 10,
                    }
                  }
                  textColor="black"
                >
                  Export
                  <Icon
                    style={
                      Object {
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  />
                </WithTheme(Button)>
              </FlexGrid>
            </FlexGrid>
            <FlexGrid
              align="stretch"
              circular={false}
              container={true}
              flex={false}
              justify="stretch"
              padding={0}
              raised={false}
              style={Object {}}
              type="row"
            >
              <FlexGrid
                align="stretch"
                circular={false}
                container={true}
                flex={true}
                justify="stretch"
                padding={30}
                raised={false}
                style={Object {}}
                type="column"
              >
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Created by:
                     
                  </span>
                  Admin
                   -
                   
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": "0.8rem",
                      }
                    }
                  >
                    4/8/2019
                  </span>
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Last edited by:
                     
                  </span>
                  Aleksandra Zaryanova
                   -
                   
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": "0.8rem",
                      }
                    }
                  >
                    4/18/2019
                  </span>
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Coordinator(s):
                     
                  </span>
                  <span />
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  style={
                    Object {
                      "paddingBottom": 10,
                    }
                  }
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Coder(s):
                     
                  </span>
                  <span />
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  variant="body1"
                >
                  <span
                    style={
                      Object {
                        "color": "#7b7b7b",
                        "fontSize": 14,
                        "fontWeight": 500,
                      }
                    }
                  >
                    Type:
                     
                  </span>
                  <span>
                    Legal Scan
                  </span>
                </WithStyles(Typography)>
                <FlexGrid
                  align="stretch"
                  circular={false}
                  container={false}
                  flex={false}
                  justify="stretch"
                  padding={0}
                  raised={false}
                  style={Object {}}
                  type="column"
                />
              </FlexGrid>
              <FlexGrid
                align="flex-end"
                circular={false}
                container={true}
                flex={true}
                justify="flex-end"
                padding={30}
                raised={false}
                style={Object {}}
                type="row"
              >
                <WithTheme(Button)
                  aria-label="Code project"
                  component={[Function]}
                  listButton={true}
                  raised={false}
                  style={
                    Object {
                      "borderRadius": 3,
                      "height": 50,
                      "marginRight": 20,
                      "width": 150,
                    }
                  }
                  to={
                    Object {
                      "pathname": "/project/2/code",
                    }
                  }
                  value="Code"
                />
                <WithTheme(Button)
                  aria-label="Validate project"
                  component={[Function]}
                  listButton={true}
                  raised={false}
                  style={
                    Object {
                      "borderRadius": 3,
                      "height": 50,
                      "width": 150,
                    }
                  }
                  to={
                    Object {
                      "pathname": "/project/2/validate",
                    }
                  }
                  value="Validate"
                />
              </FlexGrid>
            </FlexGrid>
          </FlexGrid>
        </FlexGrid>
      </WithStyles(Collapse)>
    </FlexGrid>
  </FlexGrid>
  <div
    onClick={[Function]}
    style={
      Object {
        "height": 30,
      }
    }
  />
</Fragment>
`;
